utils/python/CIME/create_test.py

```diff
@@ -105,15 +114,17 @@ def __init__(self, test_names,
         # This is the only data that multiple threads will simultaneously access
         # Each test has it's own index and setting/retrieving items from a list
         # is atomic, so this should be fine to use without mutex
-        self._test_states = [ (INITIAL_PHASE, TEST_PASS_STATUS) ] * len(self._test_names)
+        for test in self._tests:
```

Comment:
Ohhh, I wish you had asked me about this. I know the data structure looks a bit odd, but there was a reason for my madness. If you look at the comment above, you can see that I wanted the update of test_states to be atomic. This is important because the parallel code below is lock-free. IE, it relies on atomic operations instead of locks to be thread-safe. Updating two dicts is not atomic.